diff --git a/package/lean/autocore/Makefile b/package/lean/autocore/Makefile
index c6cde396e..347a45203 100644
--- a/package/lean/autocore/Makefile
+++ b/package/lean/autocore/Makefile
@@ -17,7 +17,7 @@ include $(INCLUDE_DIR)/package.mk
 define Package/autocore
   TITLE:=x86/x64 auto core loadbalance script.
   MAINTAINER:=Lean
-  DEPENDS:=@TARGET_x86 +bc +lm-sensors +ethtool
+  DEPENDS:=+bc +lm-sensors +ethtool
 endef
 
 define Package/autocore/description
diff --git a/package/lean/autocore/files/autocore b/package/lean/autocore/files/autocore
index 614e21761..90f9f52dd 100755
--- a/package/lean/autocore/files/autocore
+++ b/package/lean/autocore/files/autocore
@@ -31,7 +31,7 @@ start()
 	g=${a}${b}${c}${d}${e}${f}
 
 	mkdir -p /tmp/sysinfo
-	echo $g > /tmp/sysinfo/model
+	#echo $g > /tmp/sysinfo/model
 
 	a=$(ip address | grep ^[0-9] | awk -F: '{print $2}' | sed "s/ //g" | grep '^[e]' | grep -v "@" | grep -v "\.")
 	b=$(echo "$a" | wc -l)
@@ -42,7 +42,7 @@ start()
 		ethtool -K $c tx-checksum-ip-generic on >/dev/null 2>&1 || (
 		ethtool -K $c tx-checksum-ipv4 on >/dev/null 2>&1
 		ethtool -K $c tx-checksum-ipv6 on >/dev/null 2>&1)
-		ethtool -K $c tx-scatter-gather on >/dev/null 2>&1
+		#ethtool -K $c tx-scatter-gather on >/dev/null 2>&1
 		ethtool -K $c gso on >/dev/null 2>&1
 		ethtool -K $c tso on >/dev/null 2>&1
 		ethtool -K $c ufo on >/dev/null 2>&1
diff --git a/package/lean/autocore/files/sbin/cpuinfo b/package/lean/autocore/files/sbin/cpuinfo
index 8c0527f84..64278c223 100755
--- a/package/lean/autocore/files/sbin/cpuinfo
+++ b/package/lean/autocore/files/sbin/cpuinfo
@@ -1,5 +1,5 @@
 #!/bin/sh
 
-MHz=`grep 'MHz' /proc/cpuinfo | cut -c11- |sed -n '1p'`
-TEMP=`sensors 2>/dev/null | grep 'Core 0' | cut -c12-`
+MHz=`echo "$(cat /sys/devices/system/cpu/cpu[04]/cpufreq/cpuinfo_cur_freq)/1000" | bc`
+TEMP=`echo "scale=2; $(cat /sys/class/thermal/thermal_zone0/temp)/1000" | bc`°C
 echo "$MHz MHz $TEMP "
